runtime/src/accounts_file.rs

```diff
@@ -0,0 +1,165 @@
+use {
+    crate::{
+        accounts_file::AccountsFileEntry::AppendVecFile, append_vec::*,
+        storable_accounts::StorableAccounts,
+    },
+    solana_sdk::{account::ReadableAccount, clock::Slot, hash::Hash},
+    std::{borrow::Borrow, io, path::{Path, PathBuf}},
+};
+
+#[derive(Debug)]
+pub enum AccountsFileEntry {
+    AppendVecFile(AppendVec),
+}
+
+impl AccountsFileEntry {
+    pub fn new_append_vec(file: &Path, create: bool, size: usize) -> Self {
+        AccountsFileEntry::AppendVecFile(AppendVec::new(file, create, size))
+    }
+
+    pub fn new_append_vec_from_file<P: AsRef<Path>>(
+        path: P, current_len: usize) -> io::Result<(Self, usize)> {
+        let (append_vec, len) = AppendVec::new_from_file(
+                path, current_len)?;
+        Ok((AccountsFileEntry::AppendVecFile(append_vec), len))
+    }
+
+    pub fn set_no_remove_on_drop(&mut self) {
+        match self {
+            AppendVecFile(av) => av.set_no_remove_on_drop(),
+        }
+    }
+
+    pub fn flush(&self) -> io::Result<()> {
+        match self {
+            AppendVecFile(av) => av.flush(),
+        }
+    }
+
+    pub fn reset(&self) {
+        match self {
+            AppendVecFile(av) => av.reset(),
+        }
+    }
+
+    pub fn remaining_bytes(&self) -> u64 {
+        match self {
+            AppendVecFile(av) => av.remaining_bytes(),
+        }
+    }
+
+    pub fn len(&self) -> usize {
+        match self {
+            AppendVecFile(av) => av.len(),
+        }
+    }
+
+    pub fn is_empty(&self) -> bool {
+        match self {
+            AppendVecFile(av) => av.is_empty(),
+        }
+    }
+
+    pub fn capacity(&self) -> u64 {
+        match self {
+            AppendVecFile(av) => av.capacity(),
+        }
+    }
+
+    pub fn file_name(slot: Slot, id: impl std::fmt::Display) -> String {
+        format!("{slot}.{id}")
+    }
+
+    /// Return account metadata for the account at `offset` if its data doesn't overrun
+    /// the internal buffer. Otherwise return None. Also return the offset of the first byte
+    /// after the requested data that falls on a 64-byte boundary.
+    pub fn get_account<'a>(&'a self, offset: usize) -> Option<(StoredAccountMeta<'a>, usize)> {
```

Comment:
@jeffwashington Quick question:  I am trying to forward the get_account API, but I am getting the following warning.  Do you happen to know how I should fix it?

```


error: explicit lifetimes given in parameter types where they could be elided (or replaced with `'_` if needed by type declaration)
--
  | --> runtime/src/accounts_file.rs:76:5
  | \|
  | 76 \|     pub fn get_account<'a>(&'a self, offset: usize) -> Option<(StoredAccountMeta<'a>, usize)> {
  | \|     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  | \|
  | = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_lifetimes
  | = note: `-D clippy::needless-lifetimes` implied by `-D warnings`


```